### DISCUSSION TITLE
Any interest in a `mrsk server exec` command?

### DISCUSSION THREAD

**Discussion by nickhammond** [basecamp/kamal#371] [Ideas]
There's `mrsk app exec` which will run a command in your container but was thinking it'd be helpful to be able to run ad-hoc commands on your servers as well. For instance, when a security issue comes out and an upgrade needs to be applied at the server level. This would facilitate ssh'ing into however many machines and running the command instead of doing this individually. Is this already supported somehow? Or are people usually facilitating this via Ansible/Chef/Puppet/etc?

**Comment by dhh** [comment]
Would be happy to see this explored.

**Comment by alhafoudh** [comment]
I was thinking about having some extra step for provisioning the servers using `chef` since it is ruby with `chef-solo`. Maybe as some external gem like `kamal-chef`. What do you think?

**Comment by ReDev1L** [comment]
related:
https://github.com/basecamp/kamal/discussions/112#discussioncomment-5332424

**Comment by nickhammond** [comment]
Opened up a PR to run one-off `kamal server exec` commands, I didn't want to go down the interactive route if that wasn't really needed.

https://github.com/basecamp/kamal/pull/646

**Comment by abuaboud** [comment]
We have been using Kamal on production for over 6 months now. This is so much needed! 

Our use case mainly involves using it to debug issues, like when the server disk becomes full, and we need to run something on all hosts to clean up.

Thank you @nickhammond for opening PR

**Comment by nickhammond** [comment]
PR merged so this should be available in the next release https://github.com/basecamp/kamal/pull/646.
